lib/galaxy/tools/deps/conda_util.py

```diff
@@ -0,0 +1,305 @@
+import functools
+import hashlib
+import os.path
+import re
+import shutil
+from sys import platform as _platform
+import tempfile
+
+import six
+import yaml
+
+from ..deps import commands
+
+# Not sure there are security concerns, lets just fail fast if we are going
+# break shell commands we are building.
+SHELL_UNSAFE_PATTERN = re.compile(r"[\s\"']")
+
+IS_OS_X = _platform == "darwin"
+
+# BSD 3-clause
+CONDA_LICENSE = "http://docs.continuum.io/anaconda/eula"
+
+
+def conda_link():
+    if IS_OS_X:
+        url = "https://repo.continuum.io/miniconda/Miniconda-latest-MacOSX-x86_64.sh"
+    else:
+        url = "https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh"
+    return url
+
+
+def find_conda_prefix(conda_prefix=None):
+    """ If supplied conda_prefix is not set, default to the default location
+    for Miniconda installs.
+    """
+    if conda_prefix is None:
+        return os.path.join(os.path.expanduser("~"), "miniconda2")
+    return conda_prefix
+
+
+class CondaContext(object):
+
+    def __init__(self, conda_prefix=None, conda_exec=None,
+                 shell_exec=None, debug=False, ensure_channels=''):
+        if conda_prefix is None:
+            conda_prefix = find_conda_prefix(conda_prefix)
+        self.conda_prefix = conda_prefix
+        if conda_exec is None:
+            conda_exec = self._bin("conda")
+        self.conda_exec = conda_exec
+        self.debug = debug
+        self.shell_exec = shell_exec or commands.shell
+        if ensure_channels:
+            if not isinstance(ensure_channels, list):
+                ensure_channels = [c for c in ensure_channels.split(",") if c]
+
+            changed = False
+            conda_conf = self.load_condarc()
+            if "channels" not in conda_conf:
+                conda_conf["channels"] = []
+            channels = conda_conf["channels"]
+            for channel in ensure_channels:
+                if channel not in channels:
+                    changed = True
+                    channels.append(channel)
+
+            if changed:
+                self.save_condarc(conda_conf)
+
+    def load_condarc(self):
+        condarc = self.condarc
+        if os.path.exists(condarc):
+            with open(condarc, "r") as f:
+                return yaml.safe_load(f)
+        else:
+            return {"channels": ["defaults"]}
+
+    def save_condarc(self, conf):
+        condarc = self.condarc
+        with open(condarc, "w") as f:
+            return yaml.safe_dump(conf, f)
+
+    @property
+    def condarc(self):
+        return os.path.join(os.path.expanduser("~"), ".condarc")
```

Comment:
Comments from IRC:

```
14:15:36   jmchilton | is conda on your path- I guess I should remove it from the
                     | path for that test
14:16:00     erasche | heh, yep, sure is
14:16:00   jmchilton | it will go to a local folder if conda isn't on your path    
14:16:07     erasche | that's... interesting. 
14:16:17     erasche | I wonder if that's something admins will need to be aware of
14:16:22     erasche | is there any reason to default to the system conda?
14:16:56   jmchilton | the admin gave it to galaxy so galaxy should use it ?
14:18:23     erasche | that feels galaxy centric. My view is  "The admin
                     |  installed conda at some point for some purpose, now Galaxy 
                     | is behaving in ways that might cause distributed filesystem
                     |  issues if HOME for the galaxy user isn't exported over the 
                     | network"
```